awx/main/scheduler/task_manager.py

```diff
@@ -519,20 +519,27 @@ def process_pending_tasks(self, pending_tasks):
             if not found_acceptable_queue:
                 logger.debug("{} couldn't be scheduled on graph, waiting for next cycle".format(task.log_format))
 
-    def timeout_approval_node(self):
-        workflow_approvals = WorkflowApproval.objects.filter(status='pending').prefetch_related('workflow_approval_template')
+    def timeout_approval_node(self): # Add websocket stuff for when it transitions to "timed out" (maybe a websocket_emit_status() call)
+        workflow_approvals = WorkflowApproval.objects.filter(status='pending')
         now = tz_now()
         for task in workflow_approvals:
-            # TODO: copy the timeout to the job itself at launch time, not the template
-            approval_timeout_seconds = timedelta(seconds=task.workflow_approval_template.timeout)
-            if task.workflow_approval_template.timeout == 0:
+            # TODO: copy the timeout to the job itself at launch time, not the template <---- Started to implement steps to do this, but unsure...
+            approval_timeout_seconds = timedelta(seconds=task.timeout)
+            if task.timeout == 0:
                 continue
             if (now - task.created) >= approval_timeout_seconds:
-                logger.info("The approval node {} ({}) has expired after {} seconds.".format(task.name, task.id, task.workflow_approval_template.timeout))
+                logger.info("The approval node {} ({}) has expired after {} seconds.".format(task.name, task.pk, task.timeout))
```

Comment:
We might actually consider making this event (an approval doesn't get approved or denied because they ran out of time) louder than `logger.info`.  @matburt @gamuniz what do you think?

Reply:
I agree, I think this could be a `warn`